core/execution/equity_shares.go

```diff
@@ -225,35 +275,33 @@ func (es *EquityShares) SharesFromParty(party string) num.Decimal {
 // SharesExcept returns the ratio of equity for each party on the market, except
 // the ones listed in parameter.
 func (es *EquityShares) SharesExcept(except map[string]struct{}) map[string]num.Decimal {
-	// Calculate the equity for each party and the totalEquity (the sum of all
-	// the equities)
-	var totalEquity num.Decimal
-	shares := map[string]num.Decimal{}
-	for id := range es.lps {
-		// If the party is not one of the deployed parties, we just skip.
+	shares := make(map[string]num.Decimal, len(es.lps)-len(except))
+	allTotal, allTotalV := es.totalPStake, es.totalVStake
+	all := true
+	for k := range except {
+		if v, ok := es.lps[k]; ok {
+			es.totalPStake = es.totalPStake.Sub(v.stake)
+			es.totalVStake = es.totalVStake.Sub(v.vStake)
+			all = false
+		}
+	}
+	for id, v := range es.lps {
 		if _, ok := except[id]; ok {
 			continue
 		}
 		eq, err := es.equity(id)
```

Comment:
The temporary change to `es.totalPStake` and `es.totalVStake` in prep to the call to `es.equity()` feels a bit dangerous. I wonder if instead it should just be a nice and simple "normal" function and not a method on `es?` So like:
```
func equity(v *lp, totalVStake num.Decimal) (num.Decimal, error) {
	if totalVStake.IsZero() {
		return num.DecimalZero(), nil
	}
	return v.vStake.Div(totalVStake), nil
}
```
then we also don't have to worry about the error case of the party not existing, because in both calls to it we're already iterating over `e.lps`.

Reply:
I get what you're saying, and I do agree that this looks a bit iffy. I tried to keep the actual equity function the same throughout. The only difference here is that instead of iterating over all LP's and recalculating the totals, we're just updating them on the fly. This is more optimal when updating stake (LP submission/amendment), at the end of each time window, when getting the equity shares for all parties... the only place where it's a bit messy is here.

After the time spent, and sanity lost due to this change, I'm inclined to keep it as-is.